CRM/Upgrade/Incremental/php/FourSeven.php

```diff
@@ -1232,6 +1233,37 @@ public static function pickActivityRevisionPolicy(CRM_Queue_TaskContext $ctx) {
   }
 
   /**
+   * Remove the contribution logging reports which have been broken for a very long time.
+   *
+   * @param \CRM_Queue_TaskContext $ctx
+   *
+   * @return bool
+   */
+  public static function removeContributionLoggingReports(CRM_Queue_TaskContext $ctx) {
+    if (class_exists('CRM_Report_Form_Contribute_LoggingDetail') || class_exists('CRM_Report_Form_Contribute_LoggingSummary')) {
```

Comment:
note you could argue that if only one has been deleted we should still process the other. I leaned towards early back out if any evidence of custom stuff

Reply:
+1 on this